src/examples/src/widgets/constrained-input/Basic.tsx

```diff
@@ -0,0 +1,39 @@
+import { create, tsx } from '@dojo/framework/core/vdom';
+import ConstrainedInput from '@dojo/widgets/constrained-input';
+import icache from '@dojo/framework/core/middleware/icache';
+
+const factory = create({ icache });
+
+export default factory(function Basic({ middleware: { icache } }) {
+	const value1 = icache.getOrSet('value1', '');
+	const value2 = icache.getOrSet('value2', '');
+
+	return (
+		<virtual>
+			<ConstrainedInput
+				rules={{
+					length: {
+						min: 2,
+						max: 8
+					}
+				}}
+				value={value1}
+				onValue={(value) => icache.set('value1', value)}
+				label="Length"
+			/>
+			<ConstrainedInput
```

Comment:
Should these be two different examples?

Reply:
They certainly can be. I originally had a Basic example and a "Password" example that showed how you could use it as a password input. Problem was that they were so simple I wondered if they should be combined. I'm happy to separate them back out though!

Reply:
OK, I separated the examples out! There is now a simple "Basic" example, and then a new "Password" example that shows you how to use it as a password input (even though soon we will have a PasswordInput widget.... ignore that part)